src/main/java/com/prykhodko/shop/dao/impl/OrderDaoHibernate.java

```diff
@@ -0,0 +1,53 @@
+package com.prykhodko.shop.dao.impl;
+
+import com.prykhodko.shop.dao.OrderDao;
+import com.prykhodko.shop.model.StockOnOrder;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class OrderDaoHibernate implements OrderDao {
+
+    private static final Logger logger = Logger.getLogger(OrderDaoHibernate.class);
+
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public OrderDaoHibernate(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void addOrder(StockOnOrder stockOnOrder) {
+        Session session = sessionFactory.getCurrentSession();
```

Comment:
we can simplify:
sessionFactory.getCurrentSession().save(stockOnOrder);

Reply:
done